tests/unit/modules/test_nxos_upgrade.py

```diff
@@ -0,0 +1,366 @@
+class NxosUpgradeTestCase(TestCase, LoaderModuleMockMixin):
+
+    """ Test cases for salt.modules.nxos_upgrade """
+
+    platform_list = None
+
+    @staticmethod
+    def assert_platform_upgrade(condition, platform):
+
+        """ Assert platform upgrade condition and display appropriate chassis & images upon assertion failure """
+
+        assert bool(condition), "{0}: Upgrade {1} -> {2}".format(platform.chassis, platform.cimage, platform.nimage)
+
+    def setup_loader_modules(self):
+
+        """ Define list of platforms for Unit Test """
+
+        self.platform_list = [N3KPlatform(cimage='nxos.7.0.3.F3.3.bin',
+                                          nimage='nxos.9.2.1.255.bin'),
+                              N36KPlatform(cimage='nxos.9.1.2.50.bin',
+                                           nimage='nxos.9.2.2.50.bin'),
+                              N5KPlatform(ckimage='n6000-uk9-kickstart.7.3.0.N1.1.bin',
+                                          cimage='n6000-uk9.7.3.0.N1.1.bin',
+                                          nkimage='n6000-uk9-kickstart.7.3.3.N2.1.bin',
+                                          nimage='n6000-uk9.7.3.3.N2.1.bin'),
+                              N7KPlatform(ckimage='n7000-s2-kickstart.7.3.0.D1.1.bin',
+                                          cimage='n7000-s2-dk9.7.3.0.D1.1.bin',
+                                          nkimage='n7000-s2-kickstart.8.3.1.112.gbin',
+                                          nimage='n7000-s2-dk9.8.3.1.112.gbin'),
+                              N93KPlatform(cimage='nxos.7.0.3.I7.4.bin',
+                                           nimage='nxos.7.0.3.I7.5.bin'),
+                              N93KPlatform(cimage='nxos.7.0.3.I7.5.bin',
+                                           nimage='nxos.7.0.3.I7.5.bin'),
+                              N93KLXCPlatform(cimage='nxos.7.0.3.I7.4.bin',
+                                              nimage='nxos.7.0.3.I7.5.bin'),
+                              N95KPlatform(cimage='nxos.7.0.3.I7.4.bin',
+                                           nimage='nxos.9.2.2.14.bin')]
+
+        return {nxos_upgrade: {}}
+
+    def tearDown(self):
+        del self.platform_list
+
+    @staticmethod
+    def test_check_upgrade_impact_input_validation():
+
+        """ UT: nxos_upgrade module:check_upgrade_inpact method - input validation """
+
+        result = nxos_upgrade.check_upgrade_impact('dummy-platform-image.bin', issu=1)
+        assert 'Input Error' in result
+
+    @staticmethod
+    def test_upgrade_input_validation():
+
+        """ UT: nxos_upgrade module:upgrade method - input validation """
+
+        result = nxos_upgrade.upgrade('dummy-platform-image.bin', issu=1)
+        assert 'Input Error' in result
+
+    def test_check_upgrade_impact_backend_processing_error_500(self):
+
+        """ UT: nxos_upgrade module:check_upgrade_impact method - error HTTP code 500 """
+
+        for platform in self.platform_list:
```

Comment:
What is the rational for running this test for every platform in `self.platform_list`?  If I am reading this properly `backend_processing_error_500` is defined as part of the base data object `NXOSPlatform`.  It looks like we are running the same test n number of times for each platform.

Reply:
backend_processing_error_500 can be over-ridden by any of the derived platform classes.  Since the UT method must assume it is different from the base class.

Reply:
Ok.. so it's more about possible future differences across platforms and enabling the test to automatically pick up and test the changes

Reply:
At the end of the day that is correct.  Most likely this will not be changed but if it is then the intent is this UT method will not need to be changed.